src/audio/pipeline.c

```diff
@@ -145,26 +145,56 @@ struct pipeline_walk_context {
 	/**< pipelines to be scheduled after trigger walk */
 	struct list_item pipelines;
 	/*
-	 * If this flag is set, pipeline_for_each_comp() will skip all
+	 * If this flag is set, pipeline_for_each_element() will skip all
 	 * incompletely initialised components, i.e. those, whose .pipeline ==
 	 * NULL. Such components should not be skipped during initialisation
 	 * and clean up, but they should be skipped during streaming.
 	 */
 	bool skip_incomplete;
```

Comment:
What is the main need for this change? It looks a little bit sensible.

Reply:
> It looks a little bit sensible.

It's one point, it shouldn't be true, it should be clear at first glace what is happening there. Moreover @zrombel run extended test scope for those changes and didn't find any issues, I have made some minor changes since this time (mainly more comments, but also one minor change in functional code, so maybe I ask him to do this one more).

I'm tracing some bug for multicore configuration in this code area, but I took me many time to fully understand this code. It's possible that I will need 
> generic preparing components during graph walking functions, like data synchronization for multicore system.

Nevertheless I pointed many different issue related with this code.

Reply:
@ktrzcinx thanks for the explanation. No objections on my side.